test/Microsoft.AspNetCore.SystemWebAdapters.CoreServices.Tests/SessionState/SessionIntegrationTests.cs

```diff
@@ -0,0 +1,195 @@
+        Assert.Equal(expected, actual);
+    }
+
+    [InlineData("/disabled?override=required", "ReadOnly:False")]
+    [InlineData("/disabled?override=readonly", "ReadOnly:True")]
+    [InlineData("/readonly?override=required", "ReadOnly:False")]
+    [InlineData("/default?override=disabled", "Session:null")]
+    [Theory]
+    public async Task TestOverrideSessionStateBehavior(string endpoint, string expected)
+    {
+        var actual = await GetAsync(endpoint);
+        Assert.Equal(expected, actual);
+    }
+
+    private static async Task<string> GetAsync(string endpoint)
+    {
+        using var host = await new HostBuilder()
+          .ConfigureWebHost(webBuilder =>
+          {
+              webBuilder
+                  .UseTestServer(options =>
+                  {
+                      options.AllowSynchronousIO = true;
+                  })
+                  .ConfigureServices(services =>
+                  {
+                      services.AddRouting();
+                      services.AddControllers();
+                      services.AddSystemWebAdapters()
+                        .WrapAspNetCoreSession();
+                      services.AddDistributedMemoryCache();
+                  })
+                  .Configure(app =>
+                  {
+                      app.UseRouting();
+                      app.Use((ctx, next) =>
+                      {
+                          SetOverrideSessionState(ctx);
+                          return next(ctx);
+                      });
+                      app.UseSession();
+                      app.UseSystemWebAdapters();
+                      app.UseEndpoints(endpoints =>
+                      {
+                          endpoints.MapControllers();
+
+                      });
+                  });
+          })
+          .StartAsync();
+
+        var uri = new Uri(endpoint, UriKind.Relative);
+
+        try
+        {
+            return await host.GetTestClient().GetStringAsync(uri).ConfigureAwait(false);
+        }
+        finally
+        {
+            await host.StopAsync();
+        }
+    }
+
+    private static void SetOverrideSessionState(HttpContext context)
+    {
+        string? overrideValue = context.Request.QueryString["override"];
+
+        switch (overrideValue)
+        {
+            case "disabled":
+                context.SetSessionStateBehavior(SessionStateBehavior.Disabled);
+                break;
+            case "readonly":
+                context.SetSessionStateBehavior(SessionStateBehavior.ReadOnly);
+                break;
+            case "required":
+                context.SetSessionStateBehavior(SessionStateBehavior.Required);
+                break;
+            case "default":
+                context.SetSessionStateBehavior(SessionStateBehavior.Default);
+                break;
+            default:
+                break;
+        }
+    }
+}
+
+[ApiController]
+public class SessionStatusController : ControllerBase
```

Comment:
Instead of using controllers to test this, it's easier to read and maintain to just use minimal APIs. We can assume that attributes added to controllers show up as metadata in the endpoint